src/commonmark-rules.js

```diff
@@ -111,11 +111,25 @@ rules.fencedCodeBlock = {
   replacement: function (content, node, options) {
     var className = node.firstChild.className || ''
     var language = (className.match(/language-(\S+)/) || [null, ''])[1]
+    var code = node.firstChild.textContent
+
+    var fenceChar = options.fence === '~~~' ? '~' : '`'
+    var fenceSize = 3
+    var fenceInCodeRegex = new RegExp('^' + fenceChar + '{3,}', 'gm')
+
+    var match
+    while ((match = fenceInCodeRegex.exec(code))) {
+      if (match[ 0 ].length >= fenceSize) {
+        fenceSize = match[ 0 ].length + 1
```

Comment:
Minor stylistic suggestion:

```suggestion
      if (match[0].length >= fenceSize) {
        fenceSize = match[0].length + 1
```